### PR TITLE
Update git-hg README.md

### PR DESCRIPTION
**Pull request by andrew-m-leonard**
Signed-off-by: andrew-m-leonard <andrew_m_leonard@uk.ibm.com>

### CODE REVIEW

File: git-hg/README.md [hunk 1]
```diff
@@ -4,4 +4,40 @@ These scripts are run at https://ci.adoptopenjdk.net/view/git-hg/ and are respon
 
 ## For developers
 
-TBA
+OpenJDK Source Control repositories are managed using Mercurial. The scripts within this project mirror those Mercurial repositories
+to github.com/AdoptOpenJDK/openjdk-jdk\<NN\>u , which then serves as a mirror for AdoptOpenJDK to build the Hotspot variants, as well as providing branches for adding any AdoptOpenJDK vendor "patches" needed.
+
+The OpenJDK Mercurial repositories fall under three categories:
+- JDK8 Update Repositories : hg.openjdk.java.net/jdk8u/jdk8u/(corba|hotspot|jaxp|jaxws|nashorn|jdk|langtools)
```
Comment (johnoliver) [inline git-hg/README.md]: probably look a bit better in code blocks

File: git-hg/README.md [hunk 2]
```diff
@@ -4,4 +4,40 @@ These scripts are run at https://ci.adoptopenjdk.net/view/git-hg/ and are respon
...
+```
+hg.openjdk.java.net/jdk-updates/jdk<NN>u     --->    github.com/AdoptOpenJDK/openjdk-jdk<NN>u (master)
+hg.openjdk.java.net/jdk/jdk                  --->    github.com/AdoptOpenJDK/openjdk-jdku (master)
+```
+This mirroring utilizes the "git-remote-hg" Mercurial fast importer plugin (https://github.com/felipec/git-remote-hg), and mirrors
+the Mercurial "default" branch to the "master" branch in the git repo.
+
+The AdoptOpenJDK mirrors also have two vendor branches for AdoptOpenJDK to apply any extra patches needed:
+  - "dev"      = "master"(HEAD) + "AdoptOpenJDK Patches"
+  - "release"  = "master"(latest build tag) + "AdoptOpenJDK Patches"
```
Comment (johnoliver) [inline git-hg/README.md]: is it more accurate to say latest release tag?

File: git-hg/README.md (hunk 1 above)
Comment (andrew-m-leonard) [reply git-hg/README.md]: done

File: git-hg/README.md (hunk 2 above)
Comment (andrew-m-leonard) [reply git-hg/README.md]: not every tag is a "release"

Comment (andrew-m-leonard) [reply git-hg/README.md]: jdk-11.0.4+1/2/3/4/5 ...

**Pull request merged**